backend/src/main/java/ca/bc/gov/app/validator/SubmissionValidatorService.java

```diff
@@ -68,6 +74,51 @@ public Mono<ClientSubmissionDto> validate(
     return validateSubmissionDto(request).switchIfEmpty(validateSubmissionData(request, source));
   }
 
+  public Mono<Void> validateSubmissionLimit(JwtAuthenticationToken principal) {
+    String userId = JwtPrincipalUtil.getUserId(principal);
+    return checkSubmissionLimit(userId);
+  }
+
+  public Mono<Void> checkSubmissionLimit(String userId) {
+    String provider = extractPrefix(userId);
+    LocalDateTime endTime = LocalDateTime.now();
+    LocalDateTime startTime;
+    long maxSubmissions;
+
+    if ("IDIR".equalsIgnoreCase(provider)) {
+      startTime = endTime.minus(configuration.getIdirSubmissionTimeWindow());
+      maxSubmissions = configuration.getIdirMaxSubmissions();
+    } 
+    else if ("BCSC".equalsIgnoreCase(provider) || "BCEIDBUSINESS".equalsIgnoreCase(provider)) {
+      startTime = endTime.minus(configuration.getOtherSubmissionTimeWindow());
+      maxSubmissions = configuration.getOtherMaxSubmissions();
+    } 
+    else {
+        throw new IllegalArgumentException("Invalid provider " + provider);
+    }
+
+    return submissionRepository.countSubmissionsByTimeRangeAndCreatedBy(startTime, endTime, userId)
+        .flatMap(count -> {
+            if (count >= maxSubmissions) {
+                return Mono.error(
+                        new ValidationException(
+                              List.of(new ValidationError("submissionLimit", "Submission limit exceeded")))
+                        );
+            }
+            return Mono.empty();
```

Comment:
By returning `Mono.empty`, you will **cancel** any subsequent streams that don't have a `switchIfEmpty` or `defaultIfEmpty` statement.

Reply:
Ok. I have switched to Mono<Void> to avoid prematurely terminating the stream with Mono.empty() while keeping the expected return type